### PR TITLE
Spectrum analysis

### PR DESCRIPTION
**Pull request by ElaheParham**
The spectrum analysis for DRS data.
There are functions to load the White and Dark reference, load the spectrum, crop the functions, normalize them, calculate absorbance spectrum, load the components, and nonnegative least square to calculate the concentrations of components.

### CODE REVIEW

File: spectrumAnalysis.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import numpy as np
+import tkinter as tk
+from tkinter.filedialog import askopenfilename
+from scipy.optimize import nnls
+
+# global variables for cropping wavelength
+lowerLimit=510
+upperLimit=590
+
+class spectrum:
```
Comment (dccote) [inline spectrumAnalysis.py:11]: Uppercase letters for class Spectrum.

File: spectrumAnalysis.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+def normalizeRef(Spec):
+    """divide the spectrum by its standard deviation"""
+    Spec.data=Spec.data/np.std(Spec.data)
+    return Spec
+
+def loadWhiteRef(referenceNameNothinInfront='int75_LEDON_nothingInFront.csv',
+                 whiteReferenceName = 'int75_WHITEREFERENCE.csv',
+                 skipRowsNothing=23, skipRowsWhite=23):
+    ''' returns cropped (between 500 to 600) white reference and the wavelength'''
+    RefNothingInfront = pd.read_csv (referenceNameNothinInfront,sep=',',skiprows=skipRowsNothing).to_numpy()
```
Comment (dccote) [inline spectrumAnalysis.py:58]: lowercase for variables.

File: spectrumAnalysis.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    spec = spectrum()
+    spec.data = spectrumData[:, 3:]
+    spec.wavelength = spectrumData[:, 0]
+    croppedSpectrum = cropFunction(spec)
+    return croppedSpectrum
+
+def normalizeSpectrum(spec,darkRef):
+    """returns the normalized spectrum for the data"""
+    dRefTile = np.tile(darkRef.data, (spec.data.shape[1], 1))
+    SpectrumData=spec.data-dRefTile.T
```
Comment (dccote) [inline spectrumAnalysis.py:92]: lowerCase

**Review by dccote — CHANGES_REQUESTED** [review]

File: spectrumAnalysis.py (hunk 1 above)
Comment (emilemarcotte) [reply spectrumAnalysis.py]: Just solved it!

File: spectrumAnalysis.py (hunk 2 above)
Comment (emilemarcotte) [reply spectrumAnalysis.py]: Just solved it!

File: spectrumAnalysis.py (hunk 3 above)
Comment (emilemarcotte) [reply spectrumAnalysis.py]: Just solved it!

**Pull request merged**